addons/html_builder/static/tests/helpers.js

```diff
@@ -0,0 +1,245 @@
+    });
+    patchWithCleanup(Img.prototype, {
+        loadImage: () => {},
+    });
+}
+
+function getSnippetView(snippets) {
+    const { snippet_groups, snippet_custom, snippet_structure, snippet_content } = snippets;
+    return `
+    <snippets id="snippet_groups" string="Categories">
+        ${(snippet_groups || []).join("")}
+    </snippets>
+    <snippets id="snippet_structure" string="Structure">
+        ${(snippet_structure || []).join("")}
+    </snippets>
+    <snippets id="snippet_custom" string="Custom">
+        ${(snippet_custom || []).join("")}
+    </snippets>
+    <snippets id="snippet_content" string="Inner Content">
+        ${(snippet_content || []).join("")}
+    </snippets>`;
+}
+
+function getSnippetStructure({ name, content, keywords = [], groupName, imagePreview = "" }) {
+    keywords = keywords.join(", ");
+    return `<div name="${name}" data-oe-snippet-id="123" data-o-image-preview="${imagePreview}" data-oe-keywords="${keywords}" data-o-group="${groupName}">${content}</div>`;
+}
+
+class BuilderContainer extends Component {
+    static template = xml`
+        <div class="d-flex h-100 w-100" t-ref="container">
+            <div class="o_website_preview flex-grow-1" t-ref="website_preview">
+                <div class="o_iframe_container">
+                    <iframe class="h-100 w-100" t-ref="iframe" t-on-load="onLoad"/>
+                    <div t-if="this.state.isMobile" class="o_mobile_preview_layout">
+                        <img alt="phone" src="/html_builder/static/img/phone.png"/>
+                    </div>
+                </div>
+            </div>
+            <LocalOverlayContainer localOverlay="overlayRef" identifier="env.localOverlayContainerKey"/>
+            <div t-if="state.isEditing" t-att-class="{'o_builder_sidebar_open': state.isEditing}" class="o-website-builder_sidebar border-start border-dark">
+                <Builder t-props="this.getBuilderProps()"/>
+            </div>
+        </div>`;
+    static components = { Builder, LocalOverlayContainer };
+    static props = { content: String, Plugins: Array };
+
+    setup() {
+        this.state = useState({ isMobile: false, isEditing: false });
+        this.iframeRef = useRef("iframe");
+        const originalIframeLoaded = new Promise((resolve) => {
+            this._originalIframeLoadedResolve = resolve;
+        });
+        this.iframeLoaded = new Promise((resolve) => {
+            onMounted(async () => {
+                if (isBrowserFirefox()) {
+                    await originalIframeLoaded;
+                }
+
+                const el = this.iframeRef.el;
+                el.contentDocument.body.innerHTML = `<div id="wrapwrap"><div id="wrap" class="oe_structure oe_empty" data-oe-model="ir.ui.view" data-oe-id="539" data-oe-field="arch">${this.props.content}</div></div>`;
```

Comment:
ci flag can be avoided here with:

```suggestion
setElementContent(
            el.contentDocument.body,
            markup`<div id="wrapwrap"><div id="wrap" class="oe_structure oe_empty" data-oe-model="ir.ui.view" data-oe-id="539" data-oe-field="arch">${this.props.content}</div></div>`
        );
```
With `this.props.content` becoming escaped (if necessary) as a bonus. 

Not critical as it is in tests, but better give good examples even in tests as dev tend to copy/follow existing code a lot.